src/sql/src/plan/statement/ddl.rs

```diff
@@ -798,46 +800,38 @@ pub fn plan_create_source(
             let details = ProtoPostgresSourcePublicationDetails::decode(&*details)
                 .map_err(|e| sql_err!("{}", e))?;
 
-            // Create a "catalog" of the tables in the PG details.
-            let mut tables_by_name = BTreeMap::new();
-            for table in details.tables.iter() {
-                tables_by_name
-                    .entry(table.name.clone())
-                    .or_insert_with(BTreeMap::new)
-                    .entry(table.namespace.clone())
-                    .or_insert_with(BTreeMap::new)
-                    .entry(connection.database.clone())
-                    .or_insert(table);
-            }
+            let publication_details = PostgresSourcePublicationDetails::from_proto(details)
+                .map_err(|e| sql_err!("{}", e))?;
 
-            let publication_catalog = crate::catalog::SubsourceCatalog(tables_by_name);
+            let subsource_resolver =
+                SubsourceResolver::new(&connection.database, &publication_details.tables)
+                    .expect("references validated during purification");
 
             let mut text_cols: BTreeMap<Oid, BTreeSet<String>> = BTreeMap::new();
 
             // Look up the referenced text_columns in the publication_catalog.
             for name in text_columns {
-                let (qual, col) = match name.0.split_last().expect("must have at least one element")
-                {
-                    (col, qual) => (UnresolvedItemName(qual.to_vec()), col.as_str().to_string()),
-                };
+                let (col, qual) = name.0.split_last().expect("must have at least one element");
```

Comment:
this seems like a bug (should be `(qual, col)` instead of `(col, qual)`, as it was before). I can't see why the order would have changed in the `text_columns` values